l10n_ch_payment_slip/payment_slip.py

```diff
@@ -480,10 +481,30 @@ def _draw_address(self, canvas, print_settings, initial_position, font,
         text.setFont(font.name, font.size)
         text.textOut(com_partner.name)
         text.moveCursor(0.0, font.size)
-        for line in com_partner.contact_address.split("\n"):
-            if not line:
-                continue
-            text.textLine(line)
+        pt_cpny_add = print_settings.bvr_print_cpny_address
+        
+        if pt_cpny_add == 'full':
+            for line in com_partner.contact_address.split("\n"):
+                if not line:
+                    continue
+                
+                text.textLine(line)
+        else:
+            city_line = (com_partner.zip or '') + ' ' + (com_partner.city or '') + ' ' + (com_partner.state_id.code or '')
+            if pt_cpny_add == 'first_line_only':
+                if com_partner.street:
+                    line = com_partner.street
+                    text.textLine(line)
+                    
+                text.textLine(city_line)
+                
+            if pt_cpny_add == 'second_line_only':
+                if com_partner.street2:
+                    line = com_partner.street2
+                    text.textLine(line)
+         
+                text.textLine(city_line)
```

Comment:
This seems very specific, why would you like to display `street2` instead of `street` ?